### ISSUE TITLE
Multiple websocket connections inside same instance

### ISSUE THREAD

**Issue by logscl** [Ullink/simple-slack-api#180]
Hello,

Today I saw something strange: my bot (using the 1.0.0 version of the api) received multiple times the same message.

I'm not really sure, but I tracked down the issue to this event sent on the websocket: https://api.slack.com/events/reconnect_url.

As stated in the documentation, it is an experimental event. It was added in the last version of the api, here: https://github.com/Ullink/simple-slack-api/commit/13e7eee938a786867f23f368ae800c41f7e2b40e#diff-79822ecabbc36be14bb59e1555051aeeR875

I Think that this event should be ignored right now, or at least check the current state of the session and close it before opening a new one.

I can provide logging if it helps track down the issue.

Thanks.

**Comment by bcorne** [comment]
Hi,

I need to check but the URL should be used only on reconnection. Some logs would be helpful indeed.

Best regards,

Benoit

**Comment by logscl** [comment]
Hi,

I checked again my logs and I was a little quick on my first thought :).

However, I think I found the guilty part (and I was able to reproduce the issue).

Here: https://github.com/Ullink/simple-slack-api/commit/13e7eee938a786867f23f368ae800c41f7e2b40e#diff-79822ecabbc36be14bb59e1555051aeeR321, when an exception is raised (for instance when a message cannot be parsed), the socked is nullified and not closed properly.

That problem should be quick to reproduce: On a channel where a bot is present, upload a file and then add comments to it. The bot will not be able to parse the message. You will also notice that the ping continues it's incrementation instead of going to 0 again.

Could it be the issue there ?

Thanks.

Regards,

Logan
